File: core/src/ca/on/hojat/snakegame/gloweffect/screen/base/BaseScreen.java
package ca.on.hojat.snakegame.gloweffect.screen.base;

import ca.on.hojat.snakegame.gloweffect.config.Config;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.utils.viewport.FitViewport;
import com.badlogic.gdx.utils.viewport.Viewport;

public abstract class BaseScreen implements Screen {

    protected OrthographicCamera camera;
    protected Viewport viewport;

    public BaseScreen() {
        camera = new OrthographicCamera();
        camera.setToOrtho(false, Config.WIDTH, Config.HEIGHT);
        viewport = new FitViewport(Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), camera);
        camera.position.set(viewport.getWorldWidth() / 2, viewport.getWorldHeight() / 2, 0);
    }

    @Override
    public void render(float delta) {
        camera.update();
    }
}

File: core/src/ca/on/hojat/snakegame/gloweffect/screen/base/BaseScreen.kt
package ca.on.hojat.snakegame.gloweffect.screen.base

import ca.on.hojat.snakegame.gloweffect.config.Config
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Screen
import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.utils.viewport.FitViewport


abstract class BaseScreen : Screen {

    private val camera = OrthographicCamera()

    protected val viewPort = FitViewport(Gdx.graphics.width.toFloat(), Gdx.graphics.height.toFloat(), camera)

    init {
        camera.setToOrtho(false, Config.WIDTH.toFloat(), Config.HEIGHT.toFloat())
        camera.position.set(viewPort.worldWidth / 2f, viewPort.worldHeight / 2f, 0f)
    }

    override fun render(delta: Float) {
        camera.update()
    }
}